app.py

```diff
@@ -63,23 +68,52 @@ def _event_producer(queue: Queue, build_watcher_namespace: str):
     openshift = OpenShift()
     v1_build = openshift.ocp_client.resources.get(api_version="v1", kind="Build")
     for event in v1_build.watch(namespace=build_watcher_namespace):
+        event_name = event["object"].metadata.name
+        build_reference = {"build_log_reference": None, "base_input_reference": None, "output_reference": None}
         if event["object"].status.phase != "Complete":
             _LOGGER.debug(
-                "Ignoring build event for %r - not completed phase %r",
-                event["object"].metadata.name,
-                event["object"].status.phase,
+                "Ignoring build event for %r - not completed phase %r", event_name, event["object"].status.phase
             )
             continue
+        elif event["object"].status.phase == "Failed":
+            _LOGGER.debug(
+                "Submitting base_image and build_log as build event for %r - the phase is %r",
+                event_name,
+                event["object"].status.phase,
+            )
+            build_reference["base_input_reference"] = event["object"].spec.strategy.sourceStrategy.from.name
```

Comment:
This will work just for [s2i](https://docs.openshift.com/container-platform/3.4/dev_guide/builds/build_strategies.html), but not for other strategies. Worth to create if statements - if you don't plan to support other strategies, we should at least report an error to logger so that we capture that fact (we did not post create analysis).

Reply:
Thanks for the suggestion, I will note to add that. As I mentioned in the PR body this need to be re-tested as it was not fully tested due to errors in thamos.